File: frontend/src/types/note.ts
export interface NoteEntry {
  id: string;
  userId: string;
  date: string;
  title: string;
  content: string;
  tags?: string;
  folder?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateNoteRequest {
  date: string;
  title: string;
  content: string;
  tags?: string;
  folder?: string;
}

export interface UpdateNoteRequest extends CreateNoteRequest {}

File: backend/OnboardingDiary.Domain/Entities/Note.cs
namespace OnboardingDiary.Domain.Entities;

public class Note : AuditableEntity, ISoftDeletable
{
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
    public DateTime Date { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public List<string> Tags { get; set; } = new();
    public bool IsPinned { get; set; }
    public bool IsDeleted { get; set; }
}
